test/tests/functional/pbs_preemption.py

```diff
@@ -556,3 +556,131 @@ def test_insufficient_server_rassn_select_resc(self):
+        self.scheduler.add_resource('foo')
+
+        a = {'max_run_res_soft.ncpus': "[u:PBS_GENERIC=4]"}
+        self.server.manager(MGR_CMD_SET, QUEUE, a, 'workq')
+        a = {'max_run_res_soft.ncpus': "[u:PBS_GENERIC=6]"}
+        self.server.manager(MGR_CMD_SET, QUEUE, a, 'workq2')
+
+        p = "express_queue, normal_jobs, queue_softlimits"
+        a = {'preempt_prio': p}
+        self.server.manager(MGR_CMD_SET, SCHED, a)
+        self.server.manager(MGR_CMD_SET, SCHED, {'log_events':  2047})
+
+        # Submit 4 jobs requesting 1 ncpu each in workq
+        a = {ATTR_l + '.select=1:ncpus': 1}
+        jid_list = []
+        for _ in range(4):
+            j = Job(attrs=a)
+            jid = self.server.submit(j)
+            jid_list.append(jid)
+
+        # Submit 5th job that will make all the job in workq to go over its
+        # softlimits
+        a = {ATTR_l + '.select=1:ncpus=1:foo': 4}
+        j = Job(attrs=a)
+        jid = self.server.submit(j)
+        jid_list.append(jid)
+        self.server.expect(JOB, {'job_state=R': 5})
+
+        # Submit a job in workq2 which requests for 4 ncpus and 3 foo resource
+        a = {ATTR_l + '.select=1:ncpus=4:foo': 3, ATTR_q: 'workq2'}
+        j = Job(attrs=a)
+        jid = self.server.submit(j)
+        jid_list.append(jid)
+        self.server.expect(JOB, {'job_state': 'Q'}, id=jid)
+        msg = ";Preempting this job will escalate its priority"
+        self.scheduler.log_match(jid_list[4] + msg)
+
+    def test_preemption_priority_escalation_2(self):
+        """
+        Test that scheduler does not try preempting a job that escalates its
+        preemption priority when preempted. But in this case ensure that the
+        job whose preemption priority gets escalated is one of the running
+        jobs that scheduler is yet to preempt in simulated universe.
+        """
+        # create an addition queue
+        a = {'queue_type': 'execution',
+             'started': 'True',
+             'enabled': 'True'}
+        self.server.manager(MGR_CMD_CREATE, QUEUE, a, "workq2")
+
+        a = {'resources_available.ncpus': 10}
+        self.server.manager(MGR_CMD_SET, NODE, a, id=self.mom.shortname)
+
+        a = {'type': 'long', 'flag': 'nh'}
+        self.server.manager(MGR_CMD_CREATE, RSC, a, id='foo')
+
+        a = {'resources_available.foo': 10}
+        self.server.manager(MGR_CMD_SET, NODE, a, id=self.mom.shortname)
+        self.scheduler.add_resource('foo')
+
+        a = {'max_run_res_soft.ncpus': "[u:PBS_GENERIC=5]"}
+        self.server.manager(MGR_CMD_SET, QUEUE, a, 'workq')
+        # Set a soft limit on resource foo to 0 so that all jobs requesting
+        # this resource are over soft limits.
+        a = {'max_run_res_soft.foo': "[u:PBS_GENERIC=0]"}
+        self.server.manager(MGR_CMD_SET, QUEUE, a, 'workq')
+        a = {'max_run_res_soft.ncpus': "[u:PBS_GENERIC=10]"}
+        self.server.manager(MGR_CMD_SET, QUEUE, a, 'workq2')
+
```

Comment:
Why set a limit on workq2?  You submit 1 job requesting 8 cpus.  It'll never cross its limit.

Reply:
You are right, that's not needed.